plugin/core/src/com/perl5/lang/perl/idea/formatter/PerlPreFormatter.java

```diff
@@ -352,10 +354,55 @@ else if (myPerlSettings.MAIN_FORMAT == FORCE && PerlPackageUtil.MAIN_NAMESPACE_S
     }
   }
 
+  @Override
+  public void visitCommaSequenceExpr(@NotNull PsiPerlCommaSequenceExpr o) {
+    if (!myRange.contains(o.getTextRange())) {
+      return;
+    }
+    if (myPerlSettings.OPTIONAL_COMMA == FORCE) {
+      processAddCommaEndOfExpr(o);
+    }
+    else if (myPerlSettings.OPTIONAL_COMMA == SUPPRESS) {
+      processSuppressCommaEndOfExpr(o);
+    }
+    super.visitCommaSequenceExpr(o);
+  }
+
+  protected void processAddCommaEndOfExpr(PsiElement o) {
+    boolean isFoundNewline = NEWLINE_BEFORE_RIGHT_PAREN_PATTERN.matcher(
```

Comment:
This is not how things done in IDEA. You have an AST tree and should walk around, not just check texts with regeps. I'll adjust this. Thank you.